pkg/cache/clusterqueue.go

```diff
@@ -566,6 +566,8 @@ func (c *clusterQueue) flavorInUse(flavor kueue.ResourceFlavorReference) bool {
 
 func (q *LocalQueue) resetFlavorsAndResources(cqUsage resources.FlavorResourceQuantities, cqAdmittedUsage resources.FlavorResourceQuantities) {
 	// Clean up removed flavors or resources.
+	q.Lock()
```

Comment:
Was this a drive-by fix, needed fix only for this PR?

Reply:
It's part of this feature, as I introduce mutex in the LQ object, before it wasn't the case

Reply:
so before it wasn't necessary? Why is it necessary now?

Reply:
Because now we rely on cached usage per LQ to calculate `ConsumedUsage` - we use `LocalQueue.admittedUsage` in LQ controller. Hence we need synchronization mechanism to prevent data races

Reply:
is this function called often. I'm wondering if this could result in perf regression and if should be don conditionally (only if the  AdmissionFairSharing) used.

Reply:
`resetFlavorsAndResources` is called only on Create/Update for LQ and related CQ so I wouldn't anticipate much regression

Reply:
sgtm, thanks, this is totally acceptable